nx/test/nx/defn/kernel_test.exs

```diff
@@ -23,6 +23,15 @@ defmodule Nx.Defn.KernelTest do
       assert Nx.Defn.Kernel./(1, 2) == 0.5
     end
 
+    test "comparison" do
+      assert Nx.Defn.Kernel.==(0, 0) == 1
+      assert Nx.Defn.Kernel.!=(0, 0) == 0
+      assert Nx.Defn.Kernel.>(0, 0) == 0
+      assert Nx.Defn.Kernel.>=(0, 0) == 1
+      assert Nx.Defn.Kernel.<(0, 0) == 0
+      assert Nx.Defn.Kernel.<=(0, 0) == 1
```

Comment:
Shouldn't we return Nx.tensor(0, type: {:u, 8}) and the equivalent for 1, instead?
I think it would make for more uniform return types in defn, even though the overrides would make eveything work fine...

Approving this nevertheless

Reply:
You are correct. I will tackle this.